File: code/python/7.Unbuzzer/unbuzzer.py
import RPi.GPIO as GPIO
import time
import math

buzzerPin = 11    
buttonPin = 12    

def setup():
	global p
	print ('Program is starting...')
	GPIO.setmode(GPIO.BOARD)       
	GPIO.setup(buzzerPin, GPIO.OUT)   
	GPIO.setup(buttonPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)    
	p = GPIO.PWM(buzzerPin, 1)
	p.start(0);
	
def loop():
	while True:
		if GPIO.input(buttonPin)==GPIO.LOW:
			alertor()
			print ('buzzer on ...')
		else :
			stopAlertor()
			print ('buzzer off ...')
def alertor():
	p.start(50)
	for x in range(0,361):		
		sinVal = math.sin(x * (math.pi / 180.0))		
		toneVal = 2000 + sinVal * 500	
		p.ChangeFrequency(toneVal)		
		time.sleep(0.001)
		
def stopAlertor():
	p.stop()
			
def destroy():
	GPIO.output(buzzerPin, GPIO.LOW)     
	GPIO.cleanup()                     

if __name__ == '__main__':     
	setup()
	try:
		loop()
	except KeyboardInterrupt:  
		destroy()

File: code/C/7.Unbuzzer/unbuzzer.c
#include <wiringPi.h>
#include <stdio.h>
#include <softTone.h>
#include <math.h>
#define buzzeRPin 0 //define the buzzeRPin
#define buttonPin 1 //define the buttonPin
void alertor(int pin){
	 int x;
	 double sinVal, toneVal;
	 for(x=0;x<360;x++){ // The frequency is based on the sine curve.
	 sinVal = sin(x * (M_PI / 180));
	 toneVal = 2000 + sinVal * 500;
	 softToneWrite(pin,toneVal);
	 delay(1);
	}
}
void stopAlertor(int pin){
	softToneWrite(pin,0);
}
int main(void)
{
	 if(wiringPiSetup() == -1){ //when initialize wiring failed,print message to screen
	 printf("setup wiringPi failed !");
	 return 1;
 } 
 pinMode(buzzeRPin, OUTPUT);
 pinMode(buttonPin, INPUT);
 softToneCreate(buzzeRPin);
 pullUpDnControl(buttonPin, PUD_UP); //pull up to high level
 while(1){ 
	 if(digitalRead(buttonPin) == LOW){ //button has pressed down
	 alertor(buzzeRPin); //buzzer on
	 printf("alertor on...\n");
 }
 else { //button has released 
	 stopAlertor(buzzeRPin); //buzzer off
	 printf("...buzzer off\n");
 }
 }
 return 0;
}
